File: C++/Greedy-Algorithm/Egyptian_Fraction.cpp
/* C++ program to print a fraction in Egyptian Form using Greedy Algorithm*/
#include <bits/stdc++.h>
using namespace std;
void egyptian_Fraction(int n, int d)
{
    // When Both Numerator and denominator becomes zero then we simply return;
    if (d == 0 || n == 0)
        return;
    if (d % n == 0)
    {
        cout << "1/" << d / n;
        return;
    }
    if (n % d == 0)
    {
        cout << n / d;
        return;
    }
    if (n > d)
    {
        cout << n / d << " + ";
        egyptian_Fraction(n % d, d);
        return;
    }
    int x = d / n + 1;
    cout << "1/" << x << " + ";
    egyptian_Fraction(n * x - d, d * x);
}
int main()
{
    int num, den;
    cout << "Enter Numerator:";
    cin >> num;
    cout << "\nEnter Denominator:";
    cin >> den;
    cout << "Egyptian Fraction : " << num << "/" << den << " is" << endl;
    egyptian_Fraction(num, den);
    return 0;
}

File: Python/Greedy-Algorithm/Egyptian_Fraction.py
# Python program to print a fraction in Egyptian Form using Greedy Algorithm

def egyptianFraction(n, d):
    # When Both Numerator and denominator becomes zero then we simply return;
    if d == 0 or n == 0:
        return
    if d % n == 0:
        print("1/", d // n, sep='', end='')
        return
    if n % d == 0:
        print(n // d, end='')
        return
    if n > d:
        print(n // d, " + ", end='')
        egyptianFraction(n % d, d)
        return
    x = d // n + 1
    print("1/", x, " + ", end='', sep='')
    egyptianFraction(n * x - d, d * x)

# Main Function
numerator = int(input("Enter Numerator: "))
denominator = int(input("Enter Denominator: "))
print("Egyptian Fraction representation of", numerator, "/", denominator, "is")
egyptianFraction(numerator, denominator)